pages/badges.jsx

```diff
@@ -0,0 +1,190 @@
+    },
+    {
+        id: 23,
+        title: 'Collaboration',
+        status: 'pending',
+        description: 'Lorem ipsum dolor sit amet',
+        icon: '/img/components/badge-icon/Collaborative.png',
+        icon2x: '/img/components/badge-icon/Collaborative@2x.png'
+    },
+    {
+        id: 33,
+        title: 'Creativity',
+        status: 'eligible',
+        description: 'Lorem ipsum dolor sit amet',
+        icon: '/img/components/badge-icon/Creativity.png',
+        icon2x: '/img/components/badge-icon/Creativity@2x.png'
+    },
+    {
+        id: 44,
+        title: 'Creativity',
+        status: 'eligible',
+        description: 'Lorem ipsum dolor sit amet',
+        icon: '/img/components/badge-icon/Creativity.png',
+        icon2x: '/img/components/badge-icon/Creativity@2x.png'
+    },
+    {
+        id: 15,
+        title: 'Problem Solving',
+        status: 'achieved',
+        description: '',
+        icon: '/img/components/badge-icon/ProblemSolving.png',
+        icon2x: '/img/components/badge-icon/ProblemSolving@2x.png'
+    },
+    {
+        id: 6,
+        title: 'Creativity',
+        status: 'eligible',
+        description: 'Lorem ipsum dolor sit amet',
+        icon: '/img/components/badge-icon/Creativity.png',
+        icon2x: '/img/components/badge-icon/Creativity@2x.png'
+    },
+    {
+        id: 7,
+        title: 'Creativity',
+        status: 'eligible',
+        description: 'Lorem ipsum dolor sit amet',
+        icon: '/img/components/badge-icon/Creativity.png',
+        icon2x: '/img/components/badge-icon/Creativity@2x.png'
+    },
+    {
+        id: 18,
+        title: 'Problem Solving',
+        status: 'achieved',
+        description: '',
+        icon: '/img/components/badge-icon/ProblemSolving.png',
+        icon2x: '/img/components/badge-icon/ProblemSolving@2x.png'
+    }
+];
+
+var BadgesPage = React.createClass({
+    statics: {
+        pageTitle: 'Badges',
+        pageClassName: 'badges'
+    },
+    setBadgesData: function (error, data) {
+        var parsed = this.parseBadges(data);
+        this.setState({
+            badges: parsed
+        });
+    },
+    parseBadges: function (response) {
+        // do parsing here
+        var data = [];
+
+        if (response.status === '200' && response.body && response.body.data && response.body.data.length) {
+            data = Array.from(response.body.data).map(function (badgeData) {
+                return {
+                    'title': badgeData.title || '',
+                    'status': 'Achieved',
+                    'description': badgeData.criteria || '',
+                    'icon': badgeData.image || '',
+                    'icon2x': badgeData.image || ''
+                };
+            });
+        }
+
+        return data;
+    },
+    render: function () {
+        var linkUrl = "";
+        var badges = TESTING_DATA.map(function (badge) {
+            linkUrl = '/badge/' + urlize( badge.id + '-' + badge.title );
```

Comment:
while not necessary to change here I think, it's good to know that ES6 has a new string format that reduces the need for string concatenation using `+`:
```
linkUrl = '/badge/' + urlize(`${badge.id}-${badge.title}`);
```
Here, the backticks set up a template string, and the `${...}` are variable references